core/kazoo_documents-1.0.0/src/kzd_schema.erl

```diff
@@ -0,0 +1,58 @@
+%%%-------------------------------------------------------------------
+%%% @copyright (C) 2015, 2600Hz
+%%% @doc
+%%% General schema manipulation
+%%% @end
+%%% @contributors
+%%%   Hesaam Farhang
+%%%-------------------------------------------------------------------
+-module(kzd_schema).
+
+-export([find_schema/1,
+		 properties/2,
+		 max_length/2]).
+
+-include_lib("whistle/include/wh_types.hrl").
+
+-define(SCHEMA_KEYWORDS_MAXLENGTH,            <<"maxLength">>).
+
+%%% Load schema
+-spec find_schema(ne_binary()) -> api_object().
+find_schema(<<_/binary>> = Schema) ->
+    case wh_json_schema:load(Schema) of
+        {'ok', SchemaJObj} -> SchemaJObj;
+        {'error', _E} ->
+            lager:debug("failed to find schema ~s: ~p", [Schema, _E]),
+            'undefined'
+    end.
+
+%%% Meta keywords
+%%% ===================
+-spec properties(wh_json:key(), api_object()) -> wh_json:object() | 'undefined'.
+properties(Key, Schema) ->
+	case find_schema(Schema) of
+		'undefined' ->
+			'undefined';
+		SchemaJObj ->
+			get_path(Key, SchemaJObj)
+	end.
+
+%%% Keywords
+%%% ==================================
+%% String
+-spec max_length(wh_json:key(), api_object()) -> wh_json:object() | 'undefined'.
+max_length(Key, Schema) ->
+	Properties = properties(Key, Schema),
+	wh_json:get_value(?SCHEMA_KEYWORDS_MAXLENGTH, Properties).
```

Comment:
`properties/1` can return `undefined`, and given the spec for `max_length` I think you want the `get_value` call to return either an object or `undefined`, not crash.
My thought is that `properties` should return an object, be it an empty one and `get_value` will give `api_object()`.